torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1116,7 +1116,9 @@ def print_source(self, source) -> str:
 
     def _print_Symbol(self, expr) -> str:
         assert isinstance(expr, sympy.Symbol), str(type(expr))
-
+        assert self.symbol_to_source.get(expr), (
+            f"Unknown symbol {expr} created by constraints solver"
+        )
         return self.print_source(self.symbol_to_source[expr][0])
```

Comment:
This is not optimal as you are doing a dict-lookup twice. Also python allows to remove all assertions: `python -O test.py`

What about:
source = self.symbol_to_source.get(expr)
```
if source:
  return source[0]
else:
  raise UnknownSymbolError(f"Unknown symbol {expr} created by constraints solver")
```

Reply:
The surrounding code has similar assertions (see `ShapeGuardPrinter`). Will stamp a PR if you want to put one up to upgrade all of them.

Reply:
You are referring to https://github.com/avikchaudhuri/pytorch/blob/a10a820e707191cf9bc75275841a05eabdec106e/torch/fx/experimental/symbolic_shapes.py#L1086-L1090 ?

The difference is: That `assert` looks like a "real assertion", i.e. something that should hold true and is a bug if it does not. (But TBH I don't understand the text, i.e. what the issue in that case is)   
While here you create an exception which is intended to be handled at the call site. And as @ezyang mentioned with this you can't differentiate between those and might wrongly ignore a valid one.